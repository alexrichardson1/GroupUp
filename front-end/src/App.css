* {
  background-color: #ffe5d9;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  background: #ffd7ba;
  color: #222;
}

.group {
  background: #f4f4f4;
  margin: 20px;
  padding: 30px 20px;
  border: 2px solid grey;
  cursor: pointer;
  border-radius: 15px;
}

.group h1 {
  padding: 0;
  margin: 0;
  font-weight: bold;
  align-items: center;
  justify-content: space-between;
}

.group h3 {
  padding: 0;
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.btn {
  display: inline-block;
  background: rgb(133, 130, 130);
  color: #fff;
  border: none;
  padding: 10px 20px;
  margin: 5px;
  border-radius: 5px;
  cursor: pointer;
  text-decoration: none;
  font-size: 15px;
  font-family: inherit;
}

.container {
  max-width: 1000px;
  margin: 30px auto;
  overflow: auto;
  min-height: 300px;
  border: 1px solid steelblue;
  padding: 30px;
  border-radius: 5px;
}
