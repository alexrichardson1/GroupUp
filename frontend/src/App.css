* {
  background-color: rgb(238, 237, 237);
  color: #2c434b;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  /* font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif; */
  background: #ffd7ba;
  color: #222;
}

.title {
  font-weight: bolder;
}

.group {
  background: #f4f4f4;
  margin: 20px;
  padding: 30px 20px;
  border: 2px solid grey;
  cursor: pointer;
  border-radius: 15px;
}

.group h1 {
  padding: 0;
  margin: 0;
  font-weight: bold;
  align-items: center;
  justify-content: space-between;
}

.group h3 {
  padding: 0;
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.btn {
  display: inline-block;
  background: #2b7a7b;
  color: #fff;
  border: none;
  padding: 10px 20px;
  margin: 5px;
  border-radius: 5px;
  cursor: pointer;
  text-decoration: none;
  text-align: center;
  font-size: 15px;
  font-family: inherit;
}

.btn:hover {
  display: inline-block;
  background: #2c434b;
  color: #fff;
  border: none;
  padding: 10px 20px;
  margin: 5px;
  border-radius: 5px;
  cursor: pointer;
  text-decoration: none;
  text-align: center;
  font-size: 15px;
  font-family: inherit;
}

input[type="text"] {
  padding: 10px 15px;
  margin: 8px 0;
  box-sizing: border-box;
  padding-bottom: 10px;
}

.container {
  max-width: 1000px;
  margin: 30px auto;
  overflow: auto;
  min-height: 300px;
  /* border: 1px solid steelblue; */
  padding: 30px;
  padding-top: 60px;
  border-radius: 5px;
}

.filterBox {
  text-align: left;
}

.processImage {
  align-self: center;
  padding-left: 15px;
  padding-bottom: 4px;
  background-color: rgba(0, 0, 0, 0);
  -webkit-filter: invert(100%);
  filter: invert(100%);
  /* padding-right: 10px; */
  /* height: 10px; */
}

.splitterImage {
  align-self: center;
  padding-left: 15px;
  padding-bottom: 4px;
  background-color: rgba(0, 0, 0, 0);
  -webkit-filter: invert(100%);
  filter: invert(100%);
  height: 30px;
  /* padding-right: 10px; */
  /* height: 10px; */
}

.logoImage {
  align-self: center;
  background-color: rgba(0, 0, 0, 0);
  height: 35px;
  /* padding-right: 10px; */
  /* height: 10px; */
}

.navBar {
  align-items: center;
  background-color: #2c434b;
  color: white;
}

.navbar-collapse {
  background-color: rgba(0, 0, 0, 0);
  color: white;
}

.navbar-nav {
  background-color: rgba(0, 0, 0, 0);
  color: white;
}

.navbar-brand {
  background-color: rgba(0, 0, 0, 0);
  color: white;
}

.nav-link {
  padding-left: 20px;
  padding-right: 20px;
  font-weight: bold;
  background-color: rgba(0, 0, 0, 0);
  color: white;
}

.navBar-link-special {
  font-weight: bold;
  font-size: 20px;
}

.nav-link a {
  color: white;
}

/* colours: 17252a 2b7a7b 3aafa9 def2f1*/
