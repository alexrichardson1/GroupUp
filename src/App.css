* {
  background-color: #e5e9f2;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  background: #f1f5f8;
  color: #222;
}
