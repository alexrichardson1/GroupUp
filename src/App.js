import React from "react";
import ReactDOM from "react-dom";

const App = () => {
  return <h1>You are part of no groups. Something something something</h1>;
};

export default App;
