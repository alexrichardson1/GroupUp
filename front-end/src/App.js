import React from "react";
import ReactDOM from "react-dom";
import data from "./teams";
import "bootstrap/dist/css/bootstrap.css";

import "./App.css";

const App = () => {
  return (
    <main>
      <>
        <h1>Your teams</h1>
        <div></div>
        <ul className="list-group">
          {data.map((team) => {
            const { leader, teammates, requirements, adRequirements } = team;
            return (
              <li className="list-group-item">
                <React.Fragment>
                  <h1>{leader}'s Group</h1>
                  <h2>Other Members: {teammates.map((name) => name + " ")}</h2>
                  <h3>Requirements: {requirements}</h3>
                  <h3>Additional Requiremetnts: {adRequirements}</h3>
                </React.Fragment>
              </li>
            );
          })}
        </ul>
      </>
    </main>
  );
};

export default App;
