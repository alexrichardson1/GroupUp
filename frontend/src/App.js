import React from "react";
import Groups from "./Components/Groups";
import Home from "./Components/Home";
import SelectProject from "./Components/SelectProject";
import DetailedGroup from "./Components/DetailedGroup";
import { BrowserRouter as Router, Route } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.css";
import "./App.css";
import CreateGroup from "./Components/CreateGroup";
import Login from "./Components/Login";
import SignUp from "./Components/SignUp";
import ForgotPassword from "./Components/ForgotPassword";
import data from "./Teams";

const App = () => {
  return (
    <Router>
      <div className="container">
        <Route path="/" exact component={Login}></Route>
        <Route path="/signup" exact component={SignUp}></Route>
        <Route path="/forgotpassword" exact component={ForgotPassword}></Route>
        <Route path="/home" exact component={Home}></Route>
        <Route path="/selection" exact component={SelectProject}></Route>
        <Route path="/listings" exact render={() => <Groups />}></Route>
        <Route path="/group/:id" children={<DetailedGroup />}></Route>
        <Route
          path="/listing/:id"
          children={<Groups allGroups={data} />}
        ></Route>
        <Route path="/createGroup" exact component={CreateGroup}></Route>
      </div>
    </Router>
  );
};

export default App;
