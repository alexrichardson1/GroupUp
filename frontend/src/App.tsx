import { useState } from "react";
import Groups from "components/Groups";
import Home from "components/Home";
import SelectProject from "components/SelectProject";
import DetailedGroup from "components/DetailedGroup";
import { BrowserRouter as Router, Route, useParams } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.css";
import "App.css";
import CreateGroup from "components/CreateGroup";
import Login from "components/auth/Login";
import SignUp from "components/auth/SignUp";
import ForgotPassword from "components/auth/ForgotPassword";
import Help from "components/Help";
import { UserContext } from "components/auth/UserContext";
import SavedSearches from "components/SavedSearches";
import CreateHackathon from "components/CreateHackathon";
import MyGroups from "components/MyGroups";

function GroupsFunc() {
  const { id } = useParams<{ id: string }>();
  return <Groups id={parseInt(id)} />;
}

function CreateGroupFunc() {
  const { id } = useParams<{ id: string }>();
  //@ts-ignore
  return <CreateGroup id={parseInt(id)} />;
}

const App = () => {
  const [user, setUser] = useState<string>("");
  const [email, setEmail] = useState<string>("");

  return (
    <Router>
      <UserContext.Provider value={{ user, setUser, email, setEmail }}>
        <div className="container">
          <Route path="/" exact component={Login}></Route>
          <Route path="/signup" exact component={SignUp}></Route>
          <Route
            path="/forgotpassword"
            exact
            component={ForgotPassword}
          ></Route>
          <Route path="/savedsearches" exact component={SavedSearches}></Route>
          <Route path="/myGroups" exact component={MyGroups}></Route>
          <Route path="/home" exact component={Home}></Route>
          <Route path="/selection" exact component={SelectProject}></Route>
          <Route
            path="/group/:id/:projectId"
            children={<DetailedGroup />}
          ></Route>
          <Route path="/listing/:id" children={<GroupsFunc />}></Route>
          <Route path="/createGroup/:id" children={<CreateGroupFunc />}></Route>
          <Route
            path="/createHackathon"
            exact
            component={CreateHackathon}
          ></Route>
          <Route path="/help" exact component={Help}></Route>
        </div>
      </UserContext.Provider>
    </Router>
  );
};

export default App;
