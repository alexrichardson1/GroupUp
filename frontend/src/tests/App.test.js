import renderer from "react-test-renderer";
import App from "App";

test("renders correctly", async () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
