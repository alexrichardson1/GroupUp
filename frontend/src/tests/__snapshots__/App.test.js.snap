// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <nav
      className="navBar navbar navbar-expand-lg navbar-light fixed-top"
    >
      <span
        className="navbar-brand"
        style={
          Object {
            "color": "white",
          }
        }
      >
        <img
          alt="group up logo"
          className="logoImage"
          href="/home"
          onClick={[Function]}
          src="https://i.imgur.com/w92t1xH.png"
        />
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <div
          className="mr=auto navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="navbar-brand"
            style={
              Object {
                "color": "rgb(238, 237, 237)",
              }
            }
          >
            Login
          </span>
        </div>
        <div
          className="ms-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="ml-auto navHelp navbar-brand"
            href="/help"
            id="collasible-nav-dropdown"
            onClick={[Function]}
            title="Help"
          >
            <img
              alt="help"
              className="splitterImage"
              src="/info-circle.svg"
            />
          </a>
        </div>
      </div>
    </nav>
    <h2>
      Login
    </h2>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Email
        </label>
        <input
          className="form-control is-invalid"
          name="userEmail"
          onChange={[Function]}
          placeholder="johnsmith@example.com"
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
      <a
        className="btn btn-primary"
        href="/signup"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Haven't Got An Account
      </a>
      <a
        className="btn btn-primary"
        href="/forgotpassword"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Forgotten Your Password
      </a>
    </form>
  </div>
</div>
`;
